index.js

```diff
@@ -333,9 +336,9 @@ module.exports = function (filename, opts) {
     onReady,
 
     onDrain: onLoad(function (fn) {
-      if (blocksToBeWritten.size === 0) fn()
+      if (blocksToBeWritten.size === 0 && writingBlockIndex === -1) fn()
       else {
-        const latestBlockIndex = last(blocksToBeWritten.keys())
+        const latestBlockIndex = blocksToBeWritten.size > 0 ? last(blocksToBeWritten.keys()) : writingBlockIndex
```

Comment:
Does this mean that right after control flow enters the `else` block, the "else condition" may no longer be true? (due to concurrency?)

Reply:
it is because `writingBlockIndex` is set instead of `blocksToBeWritten` if its in the middle of the write, so we need this one to figure out which one to use.